src/roblox_api/instance.rs

```diff
@@ -95,6 +95,28 @@ impl LuaInstance {
         Ok(child)
     }
 
+    fn get_full_name(&self) -> rlua::Result<String> {
+        let tree = self.tree.lock().unwrap();
+
+        let instance = tree.get_by_ref(self.id).ok_or_else(|| {
+            rlua::Error::external("Cannot call GetFullName() on a destroyed instance")
+        })?;
+
+        let mut names = vec![instance.name.clone()];
```

Comment:
Let's change this from being a `Vec<String>` to being a `Vec<&str>` to avoid copying each instance's name. Using `.as_str()` will be super helpful here.